src/browser-lib/cookie-consent/CookieConsentProvider.tsx

```diff
@@ -3,83 +3,127 @@
-  hasConsentedTo: HasConsentedTo;
-  // consent by policyName
-  hasConsentedToPolicy: HasConsentedToPolicy;
+  getConsentState: (
+    policyName: "strictly-necessary" | "embedded-content" | "statistics",
+  ) => "granted" | "denied" | "pending";
 };
 
 export const cookieConsentContext = createContext<CookieConsentContext | null>(
   null,
 );
 
 export const useCookieConsent = () => {
-  const cookieConsentsContext = useContext(cookieConsentContext);
-  if (!cookieConsentsContext) {
+  const context = useContext(cookieConsentContext);
+
+  if (!context) {
     throw new Error(
       "useCookieConsent() called outside of cookieConsentContext provider",
     );
   }
-  return cookieConsentsContext;
-};
 
-type CookieConsentProviderProps = {
-  children?: React.ReactNode;
-  __testMockValue?: CookieConsentContext;
+  return context;
 };
-const CookieConsentProvider: FC<CookieConsentProviderProps> = (props) => {
-  const { children } = props;
-  const consents = useConfirmicConsents();
 
-  const showConsentManager = () => {
-    // eslint-disable-next-line
-    // @ts-ignore
-    window.Metomic("ConsentManager:show");
-  };
+const CookieConsentContextProvider = (props: PropsWithChildren) => {
+  const { state, getConsent: getConsentState } = useOakConsent();
+  const { showBanner, openSettings: showConsentManager } = useCookieConsentUI();
 
-  const hasConsentedTo = (serviceType: ServiceType) => {
-    const policyName = servicePolicyMap[serviceType];
+  useEffect(() => {
+    if (state.requiresInteraction) {
+      showBanner();
+    }
+  }, [state.requiresInteraction, showBanner]);
 
-    return consents[policyName].state;
-  };
+  return (
+    <cookieConsentContext.Provider
+      {...props}
+      value={{ getConsentState, showConsentManager }}
+    />
+  );
+};
 
-  const hasConsentedToPolicy = (policyName: CookiePolicyName) => {
-    return consents[policyName].state;
-  };
+const CookieConsentUIProvider = ({ children }: PropsWithChildren) => {
+  const { state, logConsents } = useOakConsent();
+  const policies = state.policyConsents.map((policyConsent) => ({
+    id: policyConsent.policyId,
+    label: policyConsent.policyLabel,
+    description: policyConsent.policyDescription,
+    strictlyNecessary: policyConsent.isStrictlyNecessary,
+    parties: policyConsent.policyParties.map((party) => ({
+      name: party.name,
+      policyURL: party.url,
+    })),
+  }));
+  const currentConsents = state.policyConsents.reduce<{
+    [policyId: string]: "granted" | "denied";
+  }>((acc, policyConsent) => {
+    if (policyConsent.consentState !== "pending") {
+      acc[policyConsent.policyId] = policyConsent.consentState;
+    }
+    return acc;
+  }, {});
+  // Suppress an SSR warning around using `useLayoutEffect` on the server
+  // by only rendering the consent UI on the client
+  const [isMounted, setIsMounted] = useState(false);
+  useEffect(() => {
+    setIsMounted(true);
+  }, []);
```

Comment:
I feel like this is the number 1 solution to solving all problems with React SSR. It's almost like React was never intended as an HTML templating language?